src/layer/Layer.js

```diff
@@ -78,6 +79,27 @@ export var Layer = Evented.extend({
 		return this._map.getPane(name ? (this.options[name] || name) : this.options.pane);
 	},
 
+	// @method setPane(name : String): this
+	// The layer will be moved to another pane
+	setPane: function (name) {
+		if (!this._map) {
+			this.options.pane = name;
+			return this;
+		}
+
+		if (this instanceof L.Marker) {
+			DomUtil.remove(this._icon);
+			this.options.pane = name;
+			this.getPane().appendChild(this._icon);
+		} else {
+			this._map.getRenderer(this)._removePath(this);
+			this.options.pane = name;
+			this._renderer = this._map.getRenderer(this);
+			this.onAdd();
+		}
+		return this;
```

Comment:
Shouldn't we first check if the pane is actually differs from current?
Extra DOM operations may appear costly, e.g. on mass `setStyle`.

Reply:
I had this implemented but there was a problem with `setStyle`. In the `setStyle` function it overrides the options (and the new pane) with `Util.setOptions(this, style);` and then I was not able to check anymore if the options pane is different to the new pane.

Maybe I can check over the html parent which is the current pane, if you want this.

Reply:
> it overrides the options

You may choose to set pane before overwriting, for example.

> if you want this.

Well, you better do what you want.

In general: I suppose that this PR makes sense, but I cannot review it thoroughly atm.